lib/view_component/base.rb

```diff
@@ -178,6 +178,11 @@ def short_identifier
         @short_identifier ||= defined?(Rails.root) ? source_location.sub("#{Rails.root}/", "") : source_location
       end
 
+      # Removes the first part of the path and the extension.
+      def virtual_path
```

Comment:
What do you think about moving this to the compile step? See below: 

https://github.com/github/view_component/pull/417/files#diff-08fe1c0f29a6308ad005c21f4adcfd64R241

That way we can set the value at app boot and not on first render of the component 😄

Reply:
Sounds great, thanks :)

Reply:
@joelhawksley is this what you had in mind https://github.com/github/view_component/pull/417/commits/ecd8d63e05697ceb0b95860a73604986534b97cf ? I think because I still need that memoizing instance var I've probably got it wrong. Should it be an `attr_accessor` on the class that I set in the compile method? Kind of like `source_location`?

Reply:
Or rather in the `inherited` method, like this https://github.com/github/view_component/pull/417/commits/95c19180e2f3790ea49f807673ceb4a664660396.